### ISSUE TITLE
Help - package not find with yarn but ok with npm

### ISSUE THREAD

**Issue by Inateno** [yarnpkg/yarn#6123]
Hi,

I'm new to yarn and I create a project using yarn workspace along Lerna, I was previously using npm and everything was working, decided to try yarn to optimize the process of "install".

So far everything is great, I just have an issue with "babel-plugin-transform-decorators-legacy" and/or "babel-plugin-transform-decorators" (tried both):

```
yarn install v1.7.0
[1/4] Resolving packages...
error Couldn't find package "babel-plugin-transform-decorators-legacy" on the "npm" registry.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

I tried a simple "yarn install" (reading from packages.json) and a "yarn add" (with -W) both are failing while "npm i" works.
Maybe there is something about cache or a registry stuff npm include by default and yarn doesn't?

Thanks.

**Comment by rally25rs** [comment]
It's working for me at the moment

```
/Projects/yarn-test 🐒   yarn add babel-plugin-transform-decorators-legacy
yarn add v1.7.0
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...

success Saved lockfile.
success Saved 22 new dependencies.
```

is this still an issue for you?

**Comment by Inateno** [comment]
Still the issue yes, is there something to configure somewhere maybe?
We have a local cache server, but it's running fine with npm, maybe there is a difference on the call with yarn??

![image](https://user-images.githubusercontent.com/1068709/43078372-8fa9104a-8e8a-11e8-8604-4d7ce3eb2d5d.png)

**Comment by GvRillaer** [comment]
Same Issue here...

**Comment by Inateno** [comment]
Hey, just to say that a mate in my office didn't have the problem.

I'm not sure if there is a local-cache related stuff (npm, temps files? dunno).
But since I'm not an isolated case, I let this issue open. Sounds like a weird issue!

**Temporary solution:** For those who have this problem, just manually do "npm i *mystuff*" in the root, once the workspace is set up by yarn, it's not a problem and works fine.

**Comment by arcanis** [comment]
Likely fixed by #6413 (working on a release at the moment, will probably reach rc by tomorrow).

The error message ("Couldn't find package") is triggered when Yarn doesn't get any output from the NPM registry, which is actually quite often (there's a significant amount of requests that ends in 500).

I'm not sure exactly why it occured on this package in particular (maybe something related to their sharding?), and I can't even reproduce it myself, but looking at the symptoms (the error message, the randomness of the failure, the similarity with other errors we got on our CI due to this problem) that's what makes the most sense.

I'm going to close this issue, but feel free to reopen if you see it again.
